rmi/servercode/TransImpl/TransactionManager.java

```diff
@@ -19,9 +24,40 @@
 	//Instantiate with access to MiddleWareImpl
 	public TransactionManager() {
 		transactionCounter = 0;
-		CD = new CrashDetection(this);
+		hashKey = hashKey_index_start;
+
 		// startDetector();
+		CD = new CrashDetection(this);
 		System.out.println("Transaction Manager Started...");
+
+
+		// initialize Vault
+		ArrayList<RMHashtable> default_Maps = new ArrayList<RMHashtable>();
+		for (int i = 0; i < hashKey_index_start; i++) {
+				RMHashtable empty_table = new RMHashtable();
+				default_Maps.add(empty_table);
+		}
+		System.out.println("Creating Empty Vault...");
+		shadowVault = new HashVault(default_Maps);
+
+		// Do we have a Vault already?
```

Comment:
If the vault already exists, we copy it in. If no vault already exists, we store our new empty vault instead.

Reply:
Note that instantiated this file as "new File" won't actually create a new file it one already exists. If no file with this name exists, it will create a new file with this File object pointing to it. If a file with this name already exists though, it will instead just create a pointer to the head of this file.

Another Note: If we write to this file pointer (poitining to the head of the file) it will erase and overwrite the contents of this file as we are not in the `append` mode